app/store/stacks-node/stacks-node.reducer.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  createEntityAdapter,
+  EntityState,
+  createSlice,
+  createSelector,
+  PayloadAction,
+} from '@reduxjs/toolkit';
+
+import { RootState } from '..';
+
+export interface StacksNode {
+  url: string;
+  name: string;
+  id: string;
+}
+
+export interface StacksNodeState extends EntityState<StacksNode> {
+  selectedApiId: string;
+}
+
+const stacksNodeAdapter = createEntityAdapter<StacksNode>();
+
+const defaultNode: StacksNode = {
+  url: 'https://stacks-node-api-latest.argon.blockstack.xyz/extended',
+  name: 'Blockstack PBC node',
+  id: 'default',
+};
+const emptyInitialState = stacksNodeAdapter.getInitialState({
+  selectedApiId: 'default',
+});
+const includesPbcDefaultNodeApi = stacksNodeAdapter.addOne(emptyInitialState, defaultNode);
+
+const stacksNodeSlice = createSlice({
+  name: 'stacksNodes',
+  initialState: includesPbcDefaultNodeApi,
+  reducers: {
+    upsertStacksNodeApi: stacksNodeAdapter.upsertOne,
+    removeStacksNodeApi: (state, action) => ({
+      ...stacksNodeAdapter.removeOne({ ...state, selectedApiId: 'default' }, action),
+    }),
+    setActiveStacksNode: (state, action: PayloadAction<string>) => ({
+      ...state,
+      selectedApiId: action.payload,
+    }),
```

Comment:
I know you don't like it, but redux toolkit uses immer and as such this state is mutable, removing the need to spread state. 
```tsx
setActiveStacksNode: (state, action: PayloadAction<string>) => ({
      ...state,
      selectedApiId: action.payload,
    }),
```
could be
```tsx
setActiveStacksNode(state, action: PayloadAction<string>) {
      state.selectedApiId = action.payload,
    },
```

Reply:
I'm warming to the syntax, will look into tidying this up later.